cognite/neat/rules/models/_rules/_types.py

```diff
@@ -13,16 +57,126 @@
     BeforeValidator(lambda v: [entry.strip() for entry in v.split(",")] if isinstance(v, str) else v),
 ]
 
+Namespace_ = Annotated[
+    Namespace,
+    BeforeValidator(
+        lambda v: (
+            Namespace(TypeAdapter(HttpUrl).validate_python(v))
+            if v.endswith("#") or v.endswith("/")
+            else Namespace(TypeAdapter(HttpUrl).validate_python(f"{v}/"))
+        )
+    ),
+]
+
+Prefix = Annotated[
+    str,
+    StringConstraints(pattern=prefix_compliance_regex),
+    custom_error(
+        lambda field_name, error, value: exceptions.PrefixesRegexViolation(
+            [value], prefix_compliance_regex
+        ).to_pydantic_custom_error()
+    ),
+]
+
 ExternalID = Annotated[
     str,
     Field(min_items=1, max_items=255),
 ]
 
 Version = Annotated[
     str,
-    Field(
-        description="Data model version",
-        min_length=1,
-        max_length=43,
+    StringConstraints(pattern=version_compliance_regex),
+    custom_error(
+        lambda field_name, error, value: exceptions.VersionRegexViolation(
+            value, version_compliance_regex
+        ).to_pydantic_custom_error()
+    ),
+]
+
+
+def split_parent(value: str) -> list[ParentClass]:
```

Comment:
What about making this private and put it at the end of the file to make it clear that types is what you should import from this module?
```suggestion
def _split_parent(value: str) -> list[ParentClass]:
```